flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -283,15 +284,27 @@ public Builder uidPrefix(String newPrefix) {
         }
       }
 
+      // Find out the equality field id list based on the user-provided equality field column names.
+      List<Integer> equalityFieldIds = Lists.newArrayList();
```

Comment:
If the equality field columns are null, then what about using the equality fields from the schema by default?

Reply:
Currently, the `equalityFieldIds` value come from the `equalityFieldColumns`, and the `equalityFieldColumns` setting by user or parse from primary key of SQL. 

Usually, the `equalityFieldColumns` should equal to the `identifierFieldIds` of schema, and we don't need the default value of `equalityFieldIds`. But when user remove the setting of `equalityFieldColumns` or remove parimary key from SQL, and the `identifierFieldIds` is not null. If we use `identifierFieldIds` as default value of `equalityFieldIds`, I can't imagine what's going to happen, maybe it's not what the user wants.

Reply:
I think that Flink should default to the defined identifier field IDs for upsert behavior. If the user sets the columns explicitly, then we should check and warn if they don't match.

Reply:
done